native/java/org/jpype/JPypeContext.java

```diff
@@ -202,15 +202,15 @@ private void shutdown()
 
       // Wait for any unregistered proxies to finish so that we don't yank
       // the rug out from under them result in a segfault.
-      while (this.proxyCount.get() > 0)
-      {
-        try
-        {
-          Thread.sleep(10);
-        } catch (InterruptedException ex)
-        {
-        }
-      }
+//      while (this.proxyCount.get() > 0)
```

Comment:
Do you have any intentions of reincarnating this code again, or why not just simply delete it?

Reply:
I recommend deleting it, though I am still not sure there isn't a hole in the process.   The shutdown process is always a nightmare if there are still threads in proxy.  I don't think there is a clean way to do it as the two systems don't have great capabilities.   Java doesn't have a way to break a thread with an exception unless it is waiting on IO.   And there is no way to find out which thread belong to Python so we can stop them.

Reply:
The lack of properly killing Java threads in general is a very unsatisfying situation indeed. So it would not help to hook into thread creation (if that would be possible) and maintain a list of threads created within a proxy (to be able to shut them down somehow).